applications/dashboard/models/class.updatemodel.php

```diff
@@ -667,7 +667,7 @@ public function runStructure($captureOnly = false) {
         } finally {
             if ($userID && $userID !== Gdn::session()->UserID) {
                 Gdn::session()->start($userID, false, false);
-            } elseif (!$userID) {
+            } elseif (!$userID && null !== c('Garden.Installed', false)) {
```

Comment:
I don't really understand this statement. Looking at it, this should never be true.

`Garden.Installed` will always be `true | false` because of the fallback, and null is never strictly to either of those values.

Reply:
We have an alternative way of installing Vanilla in which case `Garden.Installed === null`. These are the tests for that method.

Reply:
```suggestion
            } elseif (!$userID && null !== c('Garden.Installed', false)) {
            // Vanilla has an alternate install method where this config value is set to null
            // When this using this alternate install method we don't have authenticators configured and can't end the session.
```